src/test/java/liuni/KeyHandlerTest.java

```diff
@@ -129,4 +132,74 @@ public void testSetupKeys_createTwitterProperties_Fail_FileWriteException() {
             Assert.fail("This exception is not expected.");
         }
     }
+
+    @Test
+    public void testSetupKeys_createTwitterProperties_FailWriterClose_FileWriteException() {
+        BufferedWriter writer = mock(BufferedWriter.class);
+        keyHandler.setWriter(writer);
+
+        try {
+            IOException e = mock(IOException.class);
+            doThrow(e).when(writer).close();
+            keyHandler.setupKeys();
+
+            verify(e).printStackTrace();
+        }
+        catch (Exception e) {
+            Assert.fail("This exception is not expected.");
+        }
+    }
+
+    @Test
+    public void testSetUpKeys_otherServices() {
```

Comment:
what is this testing?

Reply:
If you had a service other than Twitter labelled in your XML